File: 10101 Bangla Numbers/solution.java
import java.io.*;

public class Main {
    static int KUTI = 10000000, LAKH = 100000, HAJAR = 1000, SHATA = 100;
    static StringBuilder output;
	public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
        int cases = 0;
        output = new StringBuilder();
        while(st.nextToken() == StreamTokenizer.TT_NUMBER) {
            long n = (long)st.nval;
            output.append(String.format("%4d.", ++cases));
            if(n == 0)
                output.append(" 0\n");
            else {
                split(n);
                output.append('\n');
            }
		}
        System.out.print(output);
	}
	
	static void split(long n) {
        if(n >= KUTI) {
            split(n / KUTI);
            output.append(" kuti");
            n %= KUTI;
        }
        if(n >= LAKH) {
            split(n / LAKH);
            output.append(" lakh");
            n %= LAKH;
        }
        if(n >= HAJAR) {
            split(n / HAJAR);
            output.append(" hajar");
            n %= HAJAR;
        }
        if(n >= SHATA) {
            split(n / SHATA);
            output.append(" shata");
            n %= SHATA;
        }
        if(n > 0)
            output.append(' ').append(n);
    }
}

File: 10101 Bangla Numbers/solution.c
#include <stdio.h>
#define KUTI 10000000
#define LAKH 100000
#define HAJAR 1000
#define SHATA 100

void split(long n) {
    if(n >= KUTI) {
        split(n / KUTI);
        fputs(" kuti", stdout);
        n %= KUTI;
    }
    if(n >= LAKH) {
        split(n / LAKH);
        fputs(" lakh", stdout);
        n %= LAKH;
    }
    if(n >= HAJAR) {
        split(n / HAJAR);
        fputs(" hajar", stdout);
        n %= HAJAR;
    }
    if(n >= SHATA) {
        split(n / SHATA);
        fputs(" shata", stdout);
        n %= SHATA;
    }
    if(n > 0)
        printf(" %ld", n);
}

int main() {
    long n;
    int cases = 0;
    while(scanf("%ld", &n) == 1) {
        printf("%4d.", ++cases);
        if(n == 0)
            puts(" 0");
        else {
            split(n);
            putchar('\n');
        }
    }
    return 0;
}
